src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/AppointmentsService.kt

```diff
@@ -78,6 +79,12 @@ class AppointmentsService(
     val appointmentList = mutableListOf<Appointment>()
     appList.forEach {
       val appointment: Appointment?
+      val duration: Duration? = try {
+        it.duration?.let { it1 -> Duration.parseIsoStringOrNull(it1) }
+      } catch (ex: IllegalArgumentException) {
+        null
```

Comment:
Can you add a warning log line here so we can track if this is happening

Reply:
Duration.parseIsoStringOrNull() takes care if any invalid data comes in and returns null when  format not supported. so i have changed to Duration.parseIsoString() so this will throw illegalArgumentException when format not supported and this will also log the duration value